src/lab/Graphs/StackBars/PlotStackBar.tsx

```diff
@@ -470,12 +489,23 @@ const PlotStackBar = ({
             onMouseMove={handleTooltip}
             fill={"transparent"}
             onClick={() => {
-              handleBarClick &&
-                handleBarClick(
+              if (
+                tooltipData &&
+                currentSelectedBar !==
+                  tooltipData[tooltipData.length - 1].data.date
+              ) {
+                setCurrentSelectedBar(
+                  tooltipData[tooltipData.length - 1].data.date ?? ""
+                );
+                handleBarClick?.(
                   tooltipData
                     ? tooltipData[tooltipData.length - 1].data.id ?? null
                     : null
                 );
+              } else {
+                setCurrentSelectedBar(NaN);
+                handleBarClick?.("");
```

Comment:
this syntax is a bit weird but I guess it's fine

Reply:
There were two options: 1) `handleBarClick &&  handleBarClick("")` 2) `handleBarClick?.("")` . I found the second one precise and small